path/tree.go

```diff
@@ -123,6 +130,41 @@ func (t *tree) Find(path []string, index int) interface{} {
 	return nil
 }
 
+// FindPrefix searches the tree given the subpath starting from given index in
+// the path array.
+// It returns the value stored at the last matched node as well as the length of
+// the matched path segments.
+// Double wildcard will not be matched.
```

Comment:
Shouldn't it still match everything it can?

Reply:
We don't have `**` in our basepath. Why should we support it?

Reply:
Also, I realize we are not following the "best match" behavior in the Find() method. For `a/b/c` would win over `a/**/c/**/f` for a path `a/b/d/c/e/f` because the former is more explicit. It's very hard to modify Find() at this point as it is aiming at completely match a given path.

Reply:
So I would prefer we stay simple for now and make sure the basepath selection works as how Apigee has been doing.

Reply:
I don't suppose it matters. I was just hoping that the core code could be potentially refactored and reused instead of copied. Perhaps this is simpler, though.

Reply:
I added the double wildcard, which is easier than I expected, because under "best match", it's very hard to reach a doulble wildcard.